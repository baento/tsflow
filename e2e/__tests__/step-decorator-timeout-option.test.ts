import runJest from "../runJest";
import { extractSummary } from "../utils";

test("Throw when timeout is exceeded", () => {
  const result = runJest("step-decorator-timeout-option");

  const { rest } = extractSummary(result.stderr);

  expect(rest).toMatchSnapshot();
});
