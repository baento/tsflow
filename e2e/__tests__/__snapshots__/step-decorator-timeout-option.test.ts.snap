// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Throw when timeout is exceeded 1`] = `
"FAIL __tests__/index.test.ts
  Timeout Options
    ✕ Throw when timeout is reached

  ● Timeout Options › Throw when timeout is reached

    Step timed out after 250ms

      59 |
      60 | const timeout = (ms: number) => {
    > 61 |   return new Promise((_, reject) => setTimeout(() => reject(new Error(\`Step timed out after \${ms}ms\`)), ms));
         |                                                             ^
      62 | };
      63 |
      64 | const parseArguments = <T>(args: readonly Argument[]): (T | null)[] => {

      at Timeout._onTimeout (../../src/gherkin/feature.ts:61:61)"
`;
